helpers/publicarMovimientoDetalle.go

```diff
@@ -0,0 +1,190 @@
+	var err error
+
+	if movimientoProcExtObtenido, err = models.GetMovimientoProcesoExternoById(idMovProcExterno); err != nil {
+		logs.Error(err)
+		outputError = errorctrl.Error("PublicarMovimientosDetalle - models.GetMovimientoProcesoExternoById(idMovProcExterno)", err, "400")
+		return []models.MovimientoDetalle{}, outputError
+	} else {
+		if err := json.Unmarshal([]byte(movimientoProcExtObtenido.Detalle), &detalleMovProcExt); err != nil {
+			logs.Error(err)
+			outputError = errorctrl.Error("PublicarMovimientosDetalle - json.Unmarshal([]byte(movimientoProcExtObtenido.Detalle), &detalleMovProcExt)", err, "500")
+			return []models.MovimientoDetalle{}, outputError
+		}
+
+		if detalleMovProcExt["Estado"].(string) == "Publicado" {
+			err := "El movimiento ya está en estado publicado, verifique el identificador enviado"
+			outputError := errorctrl.Error("crearMovimientoDetalle - estado[\"Estado\"].(string) == \"Publicado\"", err, "500")
+			return []models.MovimientoDetalle{}, outputError
+		} else if detalleMovProcExt["Estado"].(string) != "Preliminar" {
+			err := "No se reconoce el estado del movimiento proceso externo"
+			outputError := errorctrl.Error("crearMovimientoDetalle - estado[\"Estado\"].(string) != \"Preliminar\"", err, "500")
+			return []models.MovimientoDetalle{}, outputError
+		}
+	}
+
+	if result, err := ListaRubros(idMovProcExterno); err != nil {
+		return []models.MovimientoDetalle{}, err
+	} else {
+		if ultimosMovimientos, err := GetAllUltimos(result); err != nil {
+			logs.Error(err)
+			return []models.MovimientoDetalle{}, err
+		} else {
+			if cuentasPublicar, err := MovimientosDetalleCuentas(ultimosMovimientos, idMovProcExterno); err != nil {
+				logs.Error(err)
+				return []models.MovimientoDetalle{}, err
+			} else {
+				if registroCuentas, err := CrearMovimientosDetalle(cuentasPublicar, true); err != nil {
+					logs.Error(err)
+					return []models.MovimientoDetalle{}, err
+				} else {
+					movimientosDetalleRespuesta = registroCuentas
+				}
+			}
+		}
+	}
+
+	estadoPublicacion := "Publicado"
+	detalleMovProcExt["Estado"] = estadoPublicacion
+
+	if detalleMovProcExtActualizado, err := json.Marshal(detalleMovProcExt); err != nil {
+		logs.Error(err)
+		outputError = errorctrl.Error("crearMovimientoDetalle - json.Marshal(detalleMovProcExt)", err, "500")
+		return []models.MovimientoDetalle{}, outputError
+	} else {
+		movimientoProcExtObtenido.Detalle = string(detalleMovProcExtActualizado)
+	}
+
+	if err := models.UpdateMovimientoProcesoExternoById(movimientoProcExtObtenido); err != nil {
+		logs.Error(err)
+		outputError = errorctrl.Error("crearMovimientoDetalle - models.UpdateMovimientoProcesoExternoById(movimientoProcExtObtenido)", err, "500")
+		return []models.MovimientoDetalle{}, outputError
+	}
+
+	return movimientosDetalleRespuesta, nil
+}
+
+func ListaRubros(idMovProcExterno int) (detalleCuentasRespuesta []models.CuentasMovimientoProcesoExterno, outputError map[string]interface{}) {
```

Comment:
Falta documentar, cambiar el `...` del _suggested change_ para ello
```suggestion
// ListaRubros ...
func ListaRubros(idMovProcExterno int) (detalleCuentasRespuesta []models.CuentasMovimientoProcesoExterno, outputError map[string]interface{}) {
```